### PR TITLE
[Collections] Don't queue populateTargetTrie if there is no data

### PR DESCRIPTION
**Pull request by yim-lee**
Motivation:
`populateTargetTrie`, potentially because it's done in a background queue, causes a lot of random crashes in tests (not just in SwiftPM but other projects as well). The exception happens inside SQLite C library and it's not yet clear what the real cause is. However, since most tests do not use the package collections feature at all, `populateTargetTrie` is basically a no-op and therefore doesn't need to be queued.

Modification:
Queue `populateTargetTrie` only if there is package collection data.

This code change doesn't actually fix anything but may perhaps reduce the chance of crashing.

rdar://80840989

### CODE REVIEW

**Comment by yim-lee** [comment]
@swift-ci please smoke test

File: Sources/PackageCollections/Storage/SQLitePackageCollectionsStorage.swift [hunk 1]
```diff
@@ -801,6 +801,27 @@ final class SQLitePackageCollectionsStorage: PackageCollectionsStorage, Closable
...
+        do {
+            let numberOfCollections: Int = try self.executeStatement("SELECT COUNT(*) FROM \(Self.packageCollectionsTableName);") { statement in
+                let row = try statement.step()
+                guard let count = row?.int(at: 0) else {
+                    throw StringError("Failed to get count of \(Self.packageCollectionsTableName) table")
+                }
+                return count
+            }
+            // No collections means no data, so no need to populate target trie
+            guard numberOfCollections > 0 else {
```
Comment (tomerd) [inline Sources/PackageCollections/Storage/SQLitePackageCollectionsStorage.swift:814]: is there an edge case here for the last collection being deleted where we would want to reset the trie?

**Review by tomerd — APPROVED** [review]

File: Sources/PackageCollections/Storage/SQLitePackageCollectionsStorage.swift (hunk 1 above)
Comment (yim-lee) [reply Sources/PackageCollections/Storage/SQLitePackageCollectionsStorage.swift:814]: Adding/deleting a collection updates the trie. The method here is only called by the initializer to seed the trie before any other db updates.

**Review by neonichu — APPROVED** [review]

**Comment by tomerd** [comment]
thanks @yim-lee do you want to pull this back in to 5.5?

**Pull request merged**